lib/generators/double_entry/install/templates/migration.rb

```diff
@@ -54,9 +54,21 @@ def self.up
       t.text       "log"
       t.timestamps                             :null => false
     end
+
+    create_table "double_entry_line_metadata", :force => true do |t|
+      t.integer    "line_id",               :null => false
+      t.string     "key",     :limit => 48, :null => false
+      t.string     "value",   :limit => 64, :null => false
+      t.timestamps                          :null => false
+    end
+
+    add_index "double_entry_line_metadata", ["line_id"], :name => "lines_meta_line_id_idx"
+    add_index "double_entry_line_metadata", ["key"],     :name => "lines_meta_key_idx"
+    add_index "double_entry_line_metadata", ["value"],   :name => "lines_meta_value_idx"
```

Comment:
@pablolee in that case it probably wouldn't use the index at all, rather find all metadata rows based on key, value then join to lines based on line_id.

However, the index would be used where the we have a clause on the `lines` table:
```sql
SELECT lines.* FROM lines
JOIN double_entry_metadata ON double_entry_metadata.line_id = lines.id
WHERE key = "color" AND value = "purple"
AND lines.created_at between '2015-06-01' and '2015-07-01'
AND lines.account = 'earnings'
AND lines.code = 'item_purchase'
```
In this case it would find the matching `lines`, then query the metadata using the index.

This is all theoretical though; it must be proven in a DB